src/components/vm/disks/diskAdd.jsx

```diff
@@ -441,8 +441,15 @@ export class AddDiskModalBody extends React.Component {
         stateDelta.format = getDefaultVolumeFormat(pool);
         if (['dir', 'fs', 'netfs', 'gluster', 'vstorage'].indexOf(pool.type) > -1) {
             const volume = pool.volumes.find(vol => vol.name === value);
-            if (volume && volume.format)
+            if (volume && volume.format) {
                 stateDelta.format = volume.format;
+                if (volume.format === "iso") {
+                    this.onValueChanged("busType", "sata");
+                    this.onValueChanged("device", "cdrom");
+                } else {
+                    this.onValueChanged("device", "disk");
+                }
```

Comment:
Calling the onValueChanged inside this function, is suboptimal, as this function is just supposed to return a stateDelta as the name implies. So I actually will approve this, and hopefully we can get into cleaning up this file in the future, but rewriting this to a functional component, and use 'useEffect' hook to handle dependent state updates.

@skobyda are you fine with that?

Reply:
I agree, but I need more time to get into the "big picture" of this nice project.

Reply:
thanks for taking the time to contribute! Will leave it up to @skobyda to run some manual testing on this.

Reply:
I think this.onValueChanged needs to stay here, because onValueChanged has some additional logic when changing busType and device (e.g. for changing busType, it will regenerate target, or when changing device, it will also regenerate busType).